### PR TITLE
Add Grok JPEG 2000 toolkit to oss-fuzz

### PR DESCRIPTION
**Pull request by boxerab**
grok is a jpeg 2000 codec, one of only two actively-developed OSS jpeg 2000 toolkits, the other one being OpenJPEG.

### CODE REVIEW

**Comment by boxerab** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> memo **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> #### What to do if you already signed the CLA
> ##### Individual signers
> 
>     * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> 
> ##### Corporate signers
> 
>     * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> 
>     * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> 
> information_source **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogle%2Foss-fuzz%2Fpull%2F3456) for more info**.

@googlebot I signed it!

**Comment by boxerab** [comment]
Is there a human I can speak with about this PR ?

File: projects/grok/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2017 Google Inc.
```
Comment (Dor1s) [inline projects/grok/Dockerfile]: nit: 2020

File: projects/grok/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER boxerab@gmail.com
+RUN apt-get update && apt-get install -y make cmake g++
```
Comment (Dor1s) [inline projects/grok/Dockerfile]: why do you need `g++`? the project should be built using provided `CC` and `CXX`

File: projects/grok/build.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -eu
+# Copyright 2017 Google Inc.
```
Comment (Dor1s) [inline projects/grok/build.sh]: 2020

File: projects/grok/project.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+homepage: "https://github.com/GrokImageCompression/grok"
+primary_contact: "boxerab@gmail.com"
+fuzzing_engines:
+  - libfuzzer
+  - afl
+  - dataflow
```
Comment (Dor1s) [inline projects/grok/project.yaml]: I'd recommend removing `dataflow` engine and sanitizer for now, as they require instrumenting all the dependencies and that's why the build is failing

**Comment by Dor1s** [comment]
Regarding CLA, you commit is authored by `Aaron Boxer <aaron.boxer@collabora.com>`. Is the same email set in your GitHub account?

File: projects/grok/Dockerfile (hunk 1 above)
Comment (boxerab) [reply projects/grok/Dockerfile]: Thanks, fixed.

File: projects/grok/Dockerfile (hunk 2 above)
Comment (boxerab) [reply projects/grok/Dockerfile]: Thanks, Fixed.

File: projects/grok/build.sh (hunk 3 above)
Comment (boxerab) [reply projects/grok/build.sh]: Fixed.

**Comment by boxerab** [comment]
> Regarding CLA, you commit is authored by `Aaron Boxer <aaron.boxer@collabora.com>`. Is the same email set in your GitHub account?

Yes, that was the problem! Fixed now.

File: projects/grok/project.yaml (hunk 4 above)
Comment (boxerab) [reply projects/grok/project.yaml]: Gotcha. Thanks. Now removed.

**Comment by boxerab** [comment]
@Dor1s This project was forked from OpenJPEG in 2016. Besides quite a few bug fixes, I have focused on meta data and colour management.  The library supports end to end management of ICC profiles. And it manages meta data such as  IPTC and XMP.  So, I do have users in the digital preservation community.  Github stats say there are around 100 downloads per month.

I have recently added support for Part 15 of the standard which gives a considerable speedup of both encode and decode.

The library is one of only two actively developed open-source jpeg 2000 toolkits, the other being OpenJPEG.

**Comment by boxerab** [comment]
@Dor1s Travis was failing due to missing fuzzer scripts. I've added these in - how can I re-start the travis build ?

**Comment by Dor1s** [comment]
I've re-started Travis

**Comment by boxerab** [comment]
@Dor1s I now have a missing header file in the Travis build. How would you recommend trouble shooting this?  `grok.h` is the library header file, but compiler can't find it.

**Comment by Dor1s** [comment]
You can reproduce the failure by running the following commands in your OSS-Fuzz checkout:

```shell
python infra/helper.py build_image grok
python infra/helper.py build_fuzzers grok
```


I think your `SRC_DIR` variable might be incorrect in https://github.com/GrokImageCompression/grok/blob/master/tests/fuzzers/build_google_oss_fuzzers.sh#L20

Instead of assuming some relative path, it's recommended to rely on `$SRC` variable, i.e. your checkout will be at `$SRC/grok` and `-I<path>` arguments should be fixed with that

**Comment by boxerab** [comment]
> You can reproduce the failure by running the following commands in your OSS-Fuzz checkout:
> 
> ```shell
> python infra/helper.py build_image grok
> python infra/helper.py build_fuzzers grok
> ```
> 
> I think your `SRC_DIR` variable might be incorrect in https://github.com/GrokImageCompression/grok/blob/master/tests/fuzzers/build_google_oss_fuzzers.sh#L20
> 
> Instead of assuming some relative path, it's recommended to rely on `$SRC` variable, i.e. your checkout will be at `$SRC/grok` and `-I<path>` arguments should be fixed with that

Thanks, @Dor1s . So, you are suggesting I set  $SRC_DIR = $SRC/grok  ?

**Comment by boxerab** [comment]
@Dor1s I've fixed the build, and pushed changes to my repo.
What's the next step ?

**Comment by Dor1s** [comment]
> Thanks, @Dor1s . So, you are suggesting I set $SRC_DIR = $SRC/grok ?

I didn't debug the issue deeply enough to suggest you a concrete path. I suggest you to rely on the `$SRC` variable and use paths relative to it. What particular value you should use -- you'll have to debug. Make sure that the header is present in one of the directories which you pass to the compiler via `-I<path>` argument.

**Comment by Dor1s** [comment]
> @Dor1s I've fixed the build, and pushed changes to my repo.
> What's the next step ?

Awesome! I've just restarted Travis, will let it verify.

**Comment by boxerab** [comment]
@Dor1s Travis is happy now. What's the next step ?

File: projects/grok/project.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+homepage: "https://github.com/GrokImageCompression/grok"
+primary_contact: "boxerab@gmail.com"
+fuzzing_engines:
```
Comment (Dor1s) [inline projects/grok/project.yaml]: I'd recommend removing this section. We'll use libFuzzer, AFL, and honggfuzz by default then.

Comment (boxerab) [reply projects/grok/project.yaml]: I've removed that, thanks!

**Comment by boxerab** [comment]
@Dor1s thank you for your help !

**Pull request merged**